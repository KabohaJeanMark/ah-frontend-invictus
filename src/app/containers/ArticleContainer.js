import React, { Component } from 'react';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import ArticlePresentationComponent from '../components/ArticleComponent';
import { postArticle } from '../../redux/actions/ArticleActions';
import { articleUrl } from '../utils';

export class ArticleContainer extends Component {
  state = {
    title: "",
    description: "",
    body: "",
    tagList: ["tags"]
  };

  checkBlank = () => {
    const { title, body, description } = this.state;
    if (title === '' || body === '' || description === '') {
      toast.dismiss();
      toast.error("All fields must be filled", {
        hideProgressBar: false,
        autoClose: 3000
      });
    }
  };

  handleChange = (event) => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };
  handleSubmit = (event) => {
    event.preventDefault();
    const { title, body, description, tagList } = this.state;
    const articleDetails = { title, body, description, tagList };
    this.checkBlank();
    var headers = {};
    headers['Content-Type'] = 'application/json';
    headers['Authorization'] = `Bearer ${localStorage.getItem('user_token')}`;
    this.props.postArticle(articleUrl, articleDetails, headers);
  };

  render () {
    return (

      <ArticlePresentationComponent
        handleSubmit={this.handleSubmit}
        handleChange={this.handleChange}
      />
    );
  }
}

export default connect(null, { postArticle })(ArticleContainer);
