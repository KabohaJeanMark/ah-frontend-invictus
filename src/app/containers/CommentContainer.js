import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import Comment from '../../app/components/Comment';
import { fetchComments } from '../../redux/actions/FetchCommentAction';
import { postComment } from '../../redux/actions/CommentAction';
import FetchComment from '../../app/components/FetchComment';

export class CommentContainer extends Component {
  constructor (props) {
    super(props);

    this.state = {
      commentBody: "",
      displayComment: []
    };
  }
  componentDidMount () {
    const { fetchComments, articleId } = this.props;
    fetchComments(articleId);
  }

  componentWillReceiveProps = (nextProps) => {
    if (nextProps.comments) {
      this.setState({ displayComment: nextProps.comments.comments });
    }
  }

  emptyBody = (comment) => {
    if (comment === undefined) {
      toast.dismiss();
      toast.error("The comment body is required", {
        hideProgressBar: false,
        autoClose: 5000
      });
    }
  };

  handleChange = (event) => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  handleSubmit = (event) => {
    const { articleId } = this.props;
    event.preventDefault();
    const { commentBody } = this.state;
    var headers = {};
    headers['Content-Type'] = 'application/json';
    headers['Authorization'] = `Bearer ${localStorage.getItem('user_token')}`;
    this.emptyBody(commentBody);
    this.props.postComment(articleId, commentBody);
  };

  render () {
    const { comments } = this.props;
    return (

      <React.Fragment>
        <Comment
          handleSubmit={this.handleSubmit}
          handleChange={this.handleChange}
        />
        <FetchComment comments={comments} />
      </React.Fragment>
    );
  }
}

export const mapStateToProps = state => ({
  comment: state.postComment,
  comments: state.fetchComment.comments
});

CommentContainer.propTypes = {
  fetchComment: PropTypes.func
};

export default connect(mapStateToProps, { postComment, fetchComments })(CommentContainer);
