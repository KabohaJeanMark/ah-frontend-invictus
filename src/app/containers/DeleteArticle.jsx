import React, { Component } from "react";
import { connect } from "react-redux";
import { deleteArticleAction } from '../../redux/actions/DeleteArticle';
import DeleteArticleComponent from '../components/DeleteArticle';

export class DeleteArticleContainer extends Component {
  handleDelete = event => {
    /* eslint-disable no-alert*/
    const delArticle = window.confirm("Are you sure you want to delete this article");
    if (delArticle === true) {
      this.props.deleteArticleAction(this.props.id);
    }
  }
  render () {
    return (
      <>
        < DeleteArticleComponent
          handleDelete={this.handleDelete}
        />
      </>
    );
  }
}

export const mapStateToProps = state => ({
  deleteArticleErrors: state.deleteArticle.deleteArticleErrors,
  deleteArticleResponse: state.deleteArticle.deleteArticleResponse
});

export default connect(
  mapStateToProps,
  { deleteArticleAction }
)(DeleteArticleContainer);
