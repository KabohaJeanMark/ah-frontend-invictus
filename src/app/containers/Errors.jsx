import React, { Component } from 'react';

export default class Errors extends Component {
  constructor (props) {
    super(props);
    this.state = { errors: [] };
  }
  componentWillReceiveProps (nextProps) {
    if (nextProps.errors) {
      this.setState({
        errors: this.mapErrorsToList(nextProps)
      });
    }
  }
   mapErrorsToList = props => {
     let items = [];
     if (Array.isArray(props.errors)) {
       for (let i = 0; i < props.errors.length; i++) {
         items.push(props.errors[i]);
       }
     } else {
       Object.keys(props.errors).forEach(key => {
         for (let i = 0; i < props.errors[key].length; i++) {
           items.push(key + ': ' + props.errors[key][i]);
         }
       });
     }
     return items;
   }
   render () {
     function Repeat (props) {
       let items = [];
       for (let i = 0; i < props.errors.length; i++) {
         items.push(props.children(i, props.errors[i]));
       }
       return <div className="danger" ><ul>{items}</ul></div>;
     }
     return (
       this.state.errors.length ? (
         <Repeat errors={this.state.errors}>
           {(index, message) => <li key={index}>{message}</li>}
         </Repeat>
       ) : (null)
     );
   }
}
