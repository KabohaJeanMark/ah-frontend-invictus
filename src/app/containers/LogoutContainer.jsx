import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { logoutAction } from '../../redux/actions/LoginActions';

export class LogoutContainer extends Component {
    componentWillMount = () => {
      localStorage.clear();
      this.props.logoutAction();
    }
    render () {
      return (
        <Redirect to="/login" />
      );
    }
}

export default connect(null, { logoutAction })(LogoutContainer);
