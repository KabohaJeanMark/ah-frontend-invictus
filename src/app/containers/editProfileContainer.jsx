import React, { Component } from 'react';
import { connect } from 'react-redux';
import firebase from '../../firebase/config';
import { editProfile } from '../../redux/actions/profileActions';
import '../../assets/styles/userprofile.scss';
import NavbarComponent from '../components/Navbar';

export class EditProfileContainer extends Component {
  constructor (props) {
    super(props);
    this.state = {
      firstname: "",
      lastname: "",
      bio: "",
      image: "",
      progress: 0,
      uploadDone: false
    };
  }

  componentWillMount () {
    const { firstname, lastname, bio, image } = this.props.profile.profile;
    this.setState({ firstname, lastname, bio, image });
  }

  componentWillReceiveProps (nextProps) {
    if (nextProps.editSuccess === true) {
      const { history } = nextProps;
      history.push('/profile');
    }
  }

  onChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }
  onSubmit = (e) => {
    e.preventDefault();
    const { firstname, lastname, bio, image } = this.state;
    const newProfile = {
      firstname: firstname,
      lastname: lastname,
      bio: bio,
      image: image
    };
    this.props.editProfile(newProfile, this.props);
  }

  /* istanbul ignore next */
  uploadImage (files) {
    const fileload = firebase
      .storage()
      .ref(`images/${files[0].name}`)
      .put(files[0]);

    fileload.then(res => {
      firebase
        .storage()
        .ref(`images/${files[0].name}`)
        .getDownloadURL()
        .then(url => {
          const image = {
            image: url
          };
          this.props.editProfile(image);
        });
    });
    fileload.on("state_changed", snapshot => {
      const uploadDone = true;
      this.setState({ uploadDone });
      const progress = Math.round(
        (snapshot.bytesTransferred / snapshot.totalBytes) * 100
      );
      this.setState({ progress });
    });
  }

  render () {
    return (
      <div>
        <div className="row">
          <div className="col-md-12"><NavbarComponent /></div>
        </div>
        <div className="row">
          <div className="col-md-4" />
          <div className="col-md-4">
            <div className="image-div" >
              {this.state.image === "" ? (<img alt="profileimage" className="profile-image"
                src={require("../../assets/images/avatar2.png")} />) :
                (<img className="profile-image" alt="profileimage" src={this.state.image} />)}
              <progress value={this.state.progress} max="100" />
              <input
                type="file"
                onChange={event => this.uploadImage(event.target.files)}
                className="upload-button mt-3"
                id="profile"
              />
            </div>
            <form onSubmit={this.onSubmit}>
              <div>
                <div className="form-group">
                  <label className="">firstname</label>
                  <input type="text" name="firstname"
                    onChange={this.onChange} className="form-control"
                    defaultValue={this.state.firstname} />
                </div>
                <div className="form-group">
                  <label className="">Lastname</label>
                  <input type="text" name="lastname" className="form-control"
                    onChange={this.onChange}
                    defaultValue={this.state.lastname} />
                </div>
                <div className="form-group">
                  <label className="">Bio</label>
                  <textarea className="form-control" name="bio" id="exampleFormControlTextarea1"
                    rows="3" onChange={this.onChange} value={this.state.bio} />
                </div>
              </div>
              <div><button className="button" type="submit">Save Changes</button></div>
            </form>
          </div>
          <div className="col-md-4" />
        </div>
      </div>
    );
  }
}


export const mapStateToProps = state => ({
  profile: state.profileReducer,
  editSuccess: state.profileReducer.editSuccess
});

export default connect(mapStateToProps, { editProfile })(EditProfileContainer);
