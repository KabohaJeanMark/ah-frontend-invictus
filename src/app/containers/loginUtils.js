export const loginUtil = {
  LOGIN_UTILS: /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/
};
