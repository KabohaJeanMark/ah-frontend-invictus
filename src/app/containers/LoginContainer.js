import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { loginAction } from '../../redux/actions/LoginActions';
import Login from '../components/Login';
import { loginUtil } from './loginUtils';

export class LoginContainer extends Component {
  constructor () {
    super();
    this.state = {
      email: '',
      password: '',
      errors: {}
    };
  }
  onChange = (event) => {
    event.preventDefault();
    this.setState({ [event.target.name]: event.target.value });
  }
  checkValidEmail = (userEmail) => {
    const mailFormat = loginUtil.LOGIN_UTILS;
    if (userEmail.match(mailFormat)) {
      return true;
    }
    return false;
  };

  checkPassword = (userPassword) => {
    if (userPassword === undefined) {
      return false;
    }
    return true;
  };

  handleSubmit = (event) => {
    event.preventDefault();
    const userData = {
      user: {
        email: this.state.email,
        password: this.state.password
      }
    };
    if (this.state.email === undefined) {
      this.setState({ errors: { userEmail: 'Email required' } });
    } else if (!this.checkValidEmail(this.state.email)) {
      this.setState({ errors: { userEmail: 'invalid email format' } });
    } else if (!this.checkPassword(this.state.password)) {
      this.setState({ errors: { userPassword: 'Password field is empty' } });
    } else {
      this.props.loginAction(userData, this.props);
    }
  }
  render () {
    return (
      <Login
        handleSubmit={this.handleSubmit}
        onChange={this.onChange}
        email={this.state.email}
        password={this.state.password}
        errors={this.state.errors}
      />
    );
  }
}

LoginContainer.propTypes = {
  email: PropTypes.string,
  password: PropTypes.string,
  handleSubmit: PropTypes.func,
  onChange: PropTypes.func,
  loginAction: PropTypes.func,
  errors: PropTypes.object
};

LoginContainer.defaultProps = {
  errors: {},
  email: '',
  password: '',
  handleSubmit: () => { },
  onChange: () => { }
};
export const mapStateToProps = state => ({ login: state.auth });
export default connect(mapStateToProps, { loginAction })(LoginContainer);
