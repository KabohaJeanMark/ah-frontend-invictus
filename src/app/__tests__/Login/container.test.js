import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import { LoginContainer, mapStateToProps } from '../../containers/LoginContainer';


describe('loginContainer Component', () => {
  const wrapper = mount(<LoginContainer />);
  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should call an onChange function', () => {
    const props = {
      onChange: jest.fn(),
      emptyEmail: jest.fn(),
      handleSubmit: jest.fn(),
      checkValidEmail: jest.fn(),
      checkPassword: jest.fn(),
      loginAction: jest.fn()
    };
    const event = {
      preventDefault: () => {},
      target: {
        name: 'body',
        value: 'abc'
      }
    };
    const component = mount(
      <LoginContainer {...props} />,
    );
    const EmailInput = component.find('#UserEmail').first();
    EmailInput.simulate('change', event);
    expect(event.target.value).toEqual('abc');
  });
  it('should call a submit a form', () => {
    const props = {
      onChange: jest.fn(),
      emptyEmail: jest.fn(),
      handleSubmit: jest.fn(),
      checkValidEmail: jest.fn(),
      checkPassword: jest.fn(),
      loginAction: jest.fn()
    };
    const wrapper = mount(
      <LoginContainer {...props} />,
    );
    wrapper.setState({
      email: 'fkiryowa'
    });
    const instance = wrapper.instance();
    const checkValidEmail = jest.spyOn(instance, 'checkValidEmail');
    instance.handleSubmit({ preventDefault: jest.fn() });
    expect(checkValidEmail).toHaveBeenCalled();
  });
  it('Valid Email', () => {
    const props = {
      onChange: jest.fn(),
      emptyEmail: jest.fn(),
      handleSubmit: jest.fn(),
      checkValidEmail: jest.fn(),
      checkPassword: jest.fn(),
      loginAction: jest.fn()
    };
    const wrapper = mount(
      <LoginContainer {...props} />,
    );
    wrapper.setState({
      email: 'fkiryowa@gmail.com'
    });
    const instance = wrapper.instance();
    const checkValidEmail = jest.spyOn(instance, 'checkValidEmail');
    instance.handleSubmit({ preventDefault: jest.fn() });
    expect(checkValidEmail).toHaveBeenCalled();
  });

  it('should call a submit a form', () => {
    const props = {
      onChange: jest.fn(),
      emptyEmail: jest.fn(),
      handleSubmit: jest.fn(),
      checkValidEmail: jest.fn(),
      checkPassword: jest.fn(),
      loginAction: jest.fn()
    };
    const wrapper = mount(
      <LoginContainer {...props} />,
    );
    wrapper.setState({
      email: 'fkiryowa',
      errors: { userEmail: 'Password field is empty' }
    });
    const instance = wrapper.instance();
    const checkValidEmail = jest.spyOn(instance, 'checkValidEmail');
    instance.handleSubmit({ preventDefault: jest.fn() });
    expect(checkValidEmail).toHaveBeenCalled();
  });

  it('undefined email', () => {
    const props = {
      onChange: jest.fn(),
      emptyEmail: jest.fn(),
      handleSubmit: jest.fn(),
      checkValidEmail: jest.fn(),
      checkPassword: jest.fn(),
      loginAction: jest.fn()
    };
    const wrapper = mount(
      <LoginContainer {...props} />,
    );
    wrapper.setState({
      email: undefined,
      errors: { userEmail: 'Email required' }
    });
    const instance = wrapper.instance();
    instance.handleSubmit({ preventDefault: jest.fn() });
    expect('Email required').toEqual(instance.state.errors.userEmail);
  });
  it('Test everything is fine', () => {
    const props = {
      onChange: jest.fn(),
      emptyEmail: jest.fn(),
      handleSubmit: jest.fn(),
      checkValidEmail: jest.fn(),
      checkPassword: jest.fn(),
      loginAction: jest.fn()
    };
    const wrapper = mount(
      <LoginContainer {...props} />,
    );
    wrapper.setState({
      email: 'fkiryowa@gmail.com',
      password: 'Test@12345',
      errors: { }
    });
    const instance = wrapper.instance();
    instance.handleSubmit({ preventDefault: jest.fn() });
    expect(props.loginAction).toHaveBeenCalled();
  });
  it('Testing mapStateToProps', () => {
    const initialState = {
      auth: {}
    };
    expect(mapStateToProps(initialState)).toEqual({
      login: {}
    });
  });
});
