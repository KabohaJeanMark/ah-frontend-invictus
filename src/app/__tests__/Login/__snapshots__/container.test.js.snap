// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loginContainer Component matches snapshot 1`] = `
<LoginContainer
  email=""
  errors={Object {}}
  handleSubmit={[Function]}
  onChange={[Function]}
  password=""
>
  <Login
    email=""
    errors={Object {}}
    handleSubmit={[Function]}
    onChange={[Function]}
    password=""
  >
    <div
      className="row"
    >
      <div
        className="col-sm-4"
      />
      <div
        className="col-sm-4"
      >
        <div
          className="banner"
        >
          <img
            alt="Banner"
            className="banner-img"
            src="ah_banner.png"
          />
        </div>
        <div
          className="card"
        >
          <form
            onSubmit={[Function]}
          >
            <div
              className="form-group"
            >
              <label>
                Login Form
              </label>
              <input
                className="form-control input-field"
                id="UserEmail"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                required={true}
                type="text"
                value=""
              />
              <input
                className="form-control input-field"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                required={true}
                type="password"
                value=""
              />
              <input
                className="btn btn-primary login-btn"
                type="submit"
              />
              <p
                className="have-account"
              >
                Forgot password? 
                <a
                  href="#"
                >
                  Reset
                </a>
              </p>
            </div>
            <div
              className="loginwith"
            >
              <img
                alt="login-with"
                className="loginwith-img"
                src="loginwith.png"
              />
              <div
                className="row social-media-row"
              >
                <div
                  className="col-md-6 col-4"
                >
                  <img
                    alt="facebook-logo"
                    className="socialmedia-img facebook-img"
                    src="facebook.png"
                  />
                  <img
                    alt="google-logo"
                    className="socialmedia-img google-img"
                    src="google.png"
                  />
                  <img
                    alt="twitter-logo"
                    className="socialmedia-img twitter-img"
                    src="twitter.png"
                  />
                </div>
              </div>
            </div>
            <p
              className="have-account"
            >
              Not yet a member? 
              <a
                href="#"
              >
                Sign up
              </a>
            </p>
          </form>
        </div>
      </div>
      <div
        className="col-sm-3"
      />
    </div>
  </Login>
</LoginContainer>
`;
